core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -544,6 +549,7 @@ class KafkaServer(
   private def controlledShutdown(): Unit = {
     val socketTimeoutMs = config.controllerSocketTimeoutMs
 
+    // TODO: Handle controlled shutdown for zkBroker when we have KRaft controller.
```

Comment:
I just filed https://issues.apache.org/jira/browse/KAFKA-14447, can you add a comment here